dubbo-config/dubbo-config-dynamic/src/main/java/org/apache/dubbo/config/dynamic/support/apollo/ApolloDynamicConfiguration.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.dubbo.config.dynamic.support.apollo;
+
+import com.ctrip.framework.apollo.Config;
+import com.ctrip.framework.apollo.ConfigChangeListener;
+import com.ctrip.framework.apollo.ConfigService;
+import com.ctrip.framework.apollo.enums.PropertyChangeType;
+import com.ctrip.framework.apollo.model.ConfigChange;
+import com.ctrip.framework.apollo.model.ConfigChangeEvent;
+import org.apache.dubbo.common.Constants;
+import org.apache.dubbo.common.URL;
+import org.apache.dubbo.config.dynamic.AbstractDynamicConfiguration;
+import org.apache.dubbo.config.dynamic.ConfigChangeType;
+import org.apache.dubbo.config.dynamic.ConfigType;
+import org.apache.dubbo.config.dynamic.ConfigurationListener;
+
+/**
+ *
+ */
+public class ApolloDynamicConfiguration extends AbstractDynamicConfiguration {
+    private static final String APOLLO_ENV_KEY = "env";
+    // FIXME the key?
+    private static final String APOLLO_ADDR_KEY = "";
+    private static final String APOLLO_CLUSTER_KEY = "apollo.cluster";
+    /**
+     * support two namespaces: application -> dubbo
+     */
+    private Config dubboConfig;
+    private Config appConfig;
```

Comment:
It's a good idea to support 2 namespaces: application and dubbo.

However, users might only use one namespace to store dubbo's config, which means apollo client might not load one of the 2 namespaces and will print warning messages saying `Could not load config for namespace xx...` every  minute, see [DefaultConfig](https://github.com/ctripcorp/apollo/blob/master/apollo-client/src/main/java/com/ctrip/framework/apollo/internals/DefaultConfig.java#L91)

I'm thinking to provide a new api for this kind of scenario, to get some config _optionally_, so that it won't print any message if load failed. How do you think?

Reply:
I think it's great. Will it be a feature like the CompositeConfiguration in [commons-configuration](https://commons.apache.org/proper/commons-configuration/userguide_v1.10/howto_compositeconfiguration.html#Composite_Configuration_Details)?

Reply:
> one of the 2 namespaces and will print warning messages saying Could not load config for namespace xx... every minute.

How's this feature going, anything to update?

Reply:
@chickenlj 

Please see the previous comment I just submitted, basically I think it's better to let the user pass in the namespace name, if it's not provided, then we could fallback to `application` or `dubbo`.

If this is the way we choose, then it's not necessary to use an API to get some config _optionally_, because it's kind of a mandatory configuration.